13_5_Paying_a_Invoice.md

```diff
@@ -0,0 +1,151 @@
+# 13.5: Paying a Invoice
+
+> :information_source: **NOTE:** This is a draft in progress, so that I can get some feedback from early reviewers. It is not yet ready for learning.
+
+In this chapter you will learn how to pay an invoice using `lightning-cli pay ` command.    As we did in previous chapter we already have an invoice to pay.  
+
+Check ln invoice:
+
+`lntb1m1p03ft7lpp5zve4dsgwgdxekqqq39vhgcnv6gfa2g2ktqy9lf0aph60d0388xmqdqqcqzpgsp545a9fphd8m5ayplcu8m5845cr4m0zcnyxddwv4g3zm32yprkfd4q9qy9qsq3s4y6cmyvh0qw9qm0sf80llxyyjy9xwrjds7lpkqhzv247jsm6q5me8t9e6ftquma664gz5u4a2rvs0yf4f0mlwtwfs6as5uj5djzhcqpnqlcj
+`  
+
+This invoice starts with prefix ln+tb+1m that indicates milli, so you should multi by 0.001.   That means we'll pay an invoice by 100000 satoshis.
+
+### Understanding your invoice
+
+If you want to decode your lightning invoice you should use `lightning-cli decodepay ` command that receives a string and checks and parses as specified by the [BOLT 11 specification](https://github.com/lightningnetwork/lightning-rfc/blob/master/11-payment-encoding.md).
+
+On success it returns an object with the following fields:
+
+* currency: Currency.
+* timestamp: the UNIX-style timestamp of the invoice.
+* expiry: the number of seconds this is valid after timestamp.
+* payee: the public key of the recipient.
+* payment_hash: the payment hash of the request.
+* signature: the DER-encoded signature.
+* description: Description.
+
+```
+c$lightning-cli --network=testnet decodepay lntb1m1p03ft7lpp5zve4dsgwgdxekqqq39vhgcnv6gfa2g2ktqy9lf0aph60d0388xmqdqqcqzpgsp545a9fphd8m5ayplcu8m5845cr4m0zcnyxddwv4g3zm32yprkfd4q9qy9qsq3s4y6cmyvh0qw9qm0sf80llxyyjy9xwrjds7lpkqhzv247jsm6q5me8t9e6ftquma664gz5u4a2rvs0yf4f0mlwtwfs6as5uj5djzhcqpnqlcj
+{
+   "currency": "tb",
+   "created_at": 1595191263,
+   "expiry": 3600,
+   "payee": "0302d48972ba7eef8b40696102ad114090fd4c146e381f18c7932a2a1d73566f84",
+   "msatoshi": 100000000,
+   "amount_msat": "100000000msat",
+   "description": "First LN Payment - Learning Bitcoin and Lightning from the Command line.",
+   "min_final_cltv_expiry": 40,
+   "payment_secret": "ad3a5486ed3ee9d207f8e1f743d6981d76f16264335ae6551116e2a204764b6a",
+   "features": "028200",
+   "payment_hash": "133356c10e434d9b0000895974626cd213d5215658085fa5fd0df4f6be2739b6",
+   "signature": "30450221008c2a4d636465de07141b7c1277ffe621244299c39361ef86c0b898aafa50de8102204de4eb2e7495839beeb5540a9caf543641e44d52fdfdcb7261aec29c951b215f"
+}
+```
+Check payee value is the same as identity_pubkey value seen previous chapter.
+
+**"identity_pubkey": "0302d48972ba7eef8b40696102ad114090fd4c146e381f18c7932a2a1d73566f84",**
+
```

Comment:
I note this is the same peer that we set the channel up with. Which is expected.

It might be worth a bit of commentary on this, also: (1) can people expect to see payment requests from people further out on the Lightning Network (obviously, that's in the spec, I just don't know if it's reality yet); and (2) if so do they need to do anything special to pay them. (I assume not; but if that's the case it'd be good to just mention that all.)